lib/auth/profile.js

```diff
@@ -8,13 +9,33 @@ module.exports = (endpoint) => {
 
   const request = api(endpoint);
 
-  return token => {
-    const headers = {
-      Authorization: `bearer ${token}`
-    };
-    return request(`/me`, { headers })
-      .then(response => response.json.data)
-      .catch(() => null);
+  return (token, session) => {
+
+    return Promise.resolve()
+      .then(() => {
+        if (!session.profile ||
+          (session.profile &&
+            moment(new Date()).isAfter(moment(session.profile.expiresAt).format()))) {
+
+          const headers = {
+            Authorization: `bearer ${token}`
+          };
+
+          return request(`/me`, { headers })
+            .then(response => {
+              const p = response.json.data;
+              p.expiresAt = moment.utc(moment(new Date()).add(600, 'seconds')).valueOf();
+              return p;
```

Comment:
I would write this as:
```js
.then(({ json: { data } }) => ({
  ...data,
  expiresAt: moment.utc(moment().add(600, 'seconds')).valueOf()
}))
```
but Lenny would probably slap me

Reply:
*_however_* As per Lenny's comment, the `new Date()` bit isn't needed here either

Reply:
> but Lenny would probably slap me

@joefitter Correct. I don't mind the `...` but that destructuring... 😱

Reply:
FINE, let's merge it. I'll fix this code some time in the future. destructure all the things